.github/workflows/publish-ocr-extraction-taskrunner.yml

```
@@ -0,0 +1,109 @@
+---
+# Automatically build and publish the ocr-extraction-taskrunner to the container registry.
+# This workflow is triggered on changes to the ocr-extraction-taskrunner directory.
+
+name: Build and Publish ocr-extraction-taskrunner
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    paths:
+      - 'packages/taskrunner/**'
+      - 'packages/ocr_extraction/**'
+    branches:
+      - 'main'
+    tags:
+      - '*'
+
+jobs:
+  image-tag:
+    name: Get tagged image name
+    uses: ./.github/workflows/image-tag.yml
+    with:
+      image-name: ghcr.io/darpa-askem/ocr-extraction-taskrunner
+
+  amd64:
+    runs-on: ubuntu-24.04
+    needs: image-tag
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Login to registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Setup Java v17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 17
+          cache: gradle
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v4
+        with:
+          validate-wrappers: true
+
+      - name: Create image and push
+        uses: docker/build-push-action@v5
```

```suggestion
        uses: docker/build-push-action@v6
```

I could do this in another PR if you prefer?